Content.Server/Chat/V2/Systems/ChatAttemptHandlerSystem.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Content.Server.Administration.Managers;
+using Content.Shared.Chat.V2;
+using Content.Shared.Chat.V2.Components;
+using Content.Shared.Chat.V2.Systems;
+using Content.Shared.Ghost;
+using Content.Shared.Mobs.Systems;
+using Robust.Shared.Player;
+
+namespace Content.Server.Chat.V2.Systems;
+
+/// <summary>
+/// ChatAttemptHandlerSystem defines a set of handlers and processes for managing chat attempts sent by clients.
+/// </summary>
+public sealed class ChatAttemptHandlerSystem : EntitySystem
+{
+    private const string DeadChatFailed = "chat-system-dead-chat-failed";
+    private const string EntityNotOwnedBySender = "chat-system-entity-not-owned-by-sender";
+    private const string UnsupportedMessageType = "chat-system-unsupported-message-type";
+    private const string MissingRequiredComponent = "chat-system-missing-required-component";
+
+    [Dependency] private readonly IAdminManager _admin = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+
+    private string _deadChatFailed = "";
+    private string _entityNotOwnedBySender = "";
+    private string _unsupportedMessageType = "";
+    private string _missingRequiredComponent = "";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        _deadChatFailed = Loc.GetString(DeadChatFailed);
+        _entityNotOwnedBySender = Loc.GetString(EntityNotOwnedBySender);
+        _unsupportedMessageType = Loc.GetString(UnsupportedMessageType);
+        _missingRequiredComponent = Loc.GetString(MissingRequiredComponent);
+
+        SubscribeNetworkEvent<AttemptVerbalChatEvent>(OnAttemptChat);
+        SubscribeNetworkEvent<AttemptVisualChatEvent>(OnAttemptChat);
+        SubscribeNetworkEvent<AttemptAnnouncementEvent>(OnAttemptChat);
+        SubscribeNetworkEvent<AttemptOutOfCharacterChatEvent>(OnAttemptChat);
+    }
+
+    private void OnAttemptChat(AttemptVerbalChatEvent ev, EntitySessionEventArgs args)
```

Comment:
These handlers have similar structure but currently the typing makes them somewhat repetitive. A proposal was given to use pattern matching to help with this; this is not an area of C# I am familiar with and would leave to further optimization by more experienced coders.

The main thing here is that visual and verbal comms are split. I did attempt to combine them, but the limits of C# typing came up - there's typed error returns on validation failures that need to be specifically instantiated (C# can't just go "here's a type, can you make me an object of that type please" inside generics, which would help get code repeating down)

Do verbal and visual chat types need splitting? I think it's sensible given they work differently conceptually - the universe is modelled a bit more nicely by accepting that anything speech like is quite different to anything using body language. We don't translate emotes, for example.